webapp/src/components/Controls/DatasetSplitToggler.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { fireEvent, screen, within } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import { DatasetSplitName } from "types/api";
+import DatasetSplitToggler from "./DatasetSplitToggler";
+import { setupServer } from "msw/node";
+import {
+  getDatasetInfo,
+  getDatasetInfoWithoutEvaluationSplit,
+  getDatasetInfoWithoutTrainSplit,
+} from "mocks/api/mockDatasetInfoAPI";
+
+const onChangeMock = (value: DatasetSplitName) => jest.fn();
+const renderDataSplitToggler = (value: DatasetSplitName) =>
+  renderWithRouterAndRedux(
+    <DatasetSplitToggler value={value} onChange={onChangeMock} />,
+    { route: "/local", path: "/:jobId" }
+  );
+
+describe("DatasetInfo API", () => {
+  const handlers = [getDatasetInfo];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  test("shoud have 2 toggle buttons displayed and enabled with icons", () => {
+    renderDataSplitToggler("eval");
+    expect(screen.queryAllByRole("button")).toHaveLength(2);
+    expect(screen.queryByText("Evaluation Set")).toBeInTheDocument();
+    expect(screen.queryByTestId("SpeedIcon")).toBeInTheDocument();
+    expect(screen.queryByText("Training Set")).toBeInTheDocument();
+    expect(screen.queryByTestId("ModelTrainingIcon")).toBeInTheDocument();
+    expect(
+      screen.getAllByRole("button", { name: /Evaluation/i })[0]
+    ).toHaveAttribute("aria-pressed", "true");
+  });
+
+  test("should the toggle button Evaluation set displayed with SpeedIcon", () => {
+    renderDataSplitToggler("eval");
+    expect(within(screen.getByText("Evaluation Set")).findByRole("SpeedIcon"));
+  });
+
+  test("should the toggle button Training set displayed with ModelTrainingIcon", () => {
+    renderDataSplitToggler("eval");
+    expect(
+      within(screen.getByText("Training Set")).findByRole("ModelTrainingIcon")
+    );
+  });
+  test("should be able to test onchange updates the value property", () => {
+    renderDataSplitToggler("train");
```

Comment:
Do you mean to test that `onChange updates the value property`? But you mocked `onChange` to do nothing. And you initialized the `DataSplitToggler` with `"train"`, so the training set was already selected, and the click doesn't change anything.
```suggestion
  test("onChange updates the value property", () => {
    renderDataSplitToggler("eval");
```
On a general note, to test that something changes, you should test the initial state before changing it. Here, something like `expect(buttonEl).not.toHaveClass("Mui-selected");`. You would have noticed that the test doesn't work.